ui/src/test/java/edu/wpi/grip/ui/codegeneration/AbstractGenerationTest.java

```diff
@@ -0,0 +1,133 @@
+package edu.wpi.grip.ui.codegeneration;
+
+import edu.wpi.grip.core.Pipeline;
+import edu.wpi.grip.core.events.SourceAddedEvent;
+import edu.wpi.grip.core.operations.OperationsUtil;
+import edu.wpi.grip.core.sockets.InputSocket;
+import edu.wpi.grip.core.sockets.OutputSocket;
+import edu.wpi.grip.core.sources.ImageFileSource;
+import edu.wpi.grip.ui.codegeneration.tools.CppPipelineInterfacer;
+import edu.wpi.grip.ui.codegeneration.tools.HelperTools;
+import edu.wpi.grip.ui.codegeneration.tools.JavaPipelineInterfacer;
+import edu.wpi.grip.ui.codegeneration.tools.PipelineCreator;
+import edu.wpi.grip.ui.codegeneration.tools.PipelineGenerator;
+import edu.wpi.grip.ui.codegeneration.tools.PipelineInterfacer;
+import edu.wpi.grip.ui.codegeneration.tools.PythonPipelineInterfacer;
+import edu.wpi.grip.util.GripCoreTestModule;
+import edu.wpi.grip.util.ImageWithData;
+
+import com.google.common.eventbus.EventBus;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.experimental.categories.Category;
+import org.opencv.core.Core;
+import org.opencv.core.Mat;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.function.BooleanSupplier;
+import java.util.function.Consumer;
+
+import javax.inject.Inject;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+@Category(GenerationTest.class)
+public abstract class AbstractGenerationTest {
+  private GripCoreTestModule testModule;
+  @Inject
+  protected EventBus eventBus;
+  @Inject
+  protected Pipeline pipeline;
+  @Inject
+  protected InputSocket.Factory isf;
+  @Inject
+  protected OutputSocket.Factory osf;
+  @Inject
+  protected ImageFileSource.Factory imgfac;
+  @Inject
+  protected OperationsUtil opUtil;
+  protected PipelineGenerator gen;
+
+  static{
+    System.loadLibrary(Core.NATIVE_LIBRARY_NAME);
+  }
+  
+  @Before
+  public void setUp() {
+    testModule = new GripCoreTestModule();
+    testModule.setUp();
+    final Injector injector = Guice.createInjector(testModule);
+    injector.injectMembers(this);
+    gen = new PipelineGenerator();
+    injector.injectMembers(gen);
+  }
+
+  protected final void test(BooleanSupplier setup, Consumer<PipelineInterfacer> test, String
+      testName) {
+    assertTrue("Setup for " + testName + " reported an issue.", setup.getAsBoolean());
+    String fileName = testName;
+    gen.export(fileName);
+    JavaPipelineInterfacer jpip = new JavaPipelineInterfacer(fileName + ".java");
+    test.accept(jpip);
+    PythonPipelineInterfacer ppip = new PythonPipelineInterfacer(fileName);
+    test.accept(ppip);
+    CppPipelineInterfacer cpip = new CppPipelineInterfacer(fileName);
+    test.accept(cpip);
```

Comment:
No because parameterized tests don't work with categories. At least in the version of JUnit we are using. It gets fixed in the next version though.